script.js

```diff
@@ -1,4 +1,78 @@
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNumber = randomInteger + 1;
+  return diceNumber;
+};
+
+var gameMode = "waitingForPlayerName";
+var playerNames = [];
+var playerOneScore = [];
+var playerTwoScore = [];
+var playerOneScoreArray = [];
+var playerTwoScoreArray = [];
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (gameMode == "waitingForPlayerName") {
+    //Player One Code
+    if (input == "") {
+      return `Please Input Player 1 Name`;
+    } else {
+      playerNames.push(input);
+      gameMode = "rollDicePlayerOne";
+      return `Hello ${playerNames[0]}, welcome to Beat That! Click on Submit to Roll the dice!`;
+    }
+  }
+  if (gameMode == "rollDicePlayerOne") {
+    playerOneScoreArray.push(rollDice());
+    playerOneScoreArray.push(rollDice());
+    gameMode = "playerOneChooseOrder";
+    return `${playerNames[0]}, your first number is ${playerOneScoreArray[0]} and your second number is ${playerOneScoreArray[1]},<br>input "standard" or "reverse" to add your number up!`;
+  }
+  if (gameMode == "playerOneChooseOrder" && input == "standard") {
+    playerOneScore = "" + playerOneScoreArray[0] + playerOneScoreArray[1];
+    gameMode = "playerTwoInputName";
+    return `${playerNames[0]}! Your score is ${playerOneScoreArray[0]}${playerOneScoreArray[1]}!`;
+  }
+  if (gameMode == "playerOneChooseOrder" && input == "reverse") {
+    playerOneScore = "" + playerOneScoreArray[1] + playerOneScoreArray[0];
+    gameMode = "playerTwoInputName";
+    return `${playerNames[0]}! Your score is ${playerOneScoreArray[1]}${playerOneScoreArray[0]}!`;
```

Comment:
This could also be a helper function, in which you run if statements checking for the input, then carrying out the relevant logic